pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,309 @@ type DiscoveryReconciler struct {
+		if err := json.Unmarshal(respBody, &flavours); err != nil {
+			return nil, err
+		}
+
+		for _, flavour := range flavours {
+			klog.Infof("Flavour found: %s", flavour.Name)
+			cr := createFlavourCustomResource(flavour)
+			flavoursCR = append(flavoursCR, cr)
+		}
+
+	}
+	klog.Info("Flavours created", flavoursCR)
+	return flavoursCR, nil
+
+}
+
+// createFlavourCustomResource creates a Flavour CR from a Flavour
+func createFlavourCustomResource(flavour *nodecorev1alpha1.Flavour) *nodecorev1alpha1.Flavour {
+	flavourCR := (&nodecorev1alpha1.Flavour{
+		// There is a warning: unknown field "spec.flavour.metadata.creationTimestamp"
+		// No issues, it is caused by kubebuilder for the creationTimestamp field, common bug for this version of k8s
+		Spec: nodecorev1alpha1.FlavourSpec{
+			ProviderID: flavour.Name,
+			Type:       nodecorev1alpha1.K8S,
+			Characteristics: nodecorev1alpha1.Characteristics{
+				Cpu:    flavour.Spec.Characteristics.Cpu,
+				Memory: flavour.Spec.Characteristics.Memory,
+			},
+			Policy: nodecorev1alpha1.Policy{
+				// Check if flavour.Partitionable is not nil before setting Partitionable
+				Partitionable: func() *nodecorev1alpha1.Partitionable {
+					if flavour.Spec.Policy.Partitionable != nil {
+						return &nodecorev1alpha1.Partitionable{
+							CpuMin:     flavour.Spec.Policy.Partitionable.CpuMin,
+							MemoryMin:  flavour.Spec.Policy.Partitionable.MemoryMin,
+							CpuStep:    flavour.Spec.Policy.Partitionable.CpuStep,
+							MemoryStep: flavour.Spec.Policy.Partitionable.MemoryStep,
+						}
+					}
+					return nil
+				}(),
+				Aggregatable: func() *nodecorev1alpha1.Aggregatable {
+					if flavour.Spec.Policy.Aggregatable != nil {
+						return &nodecorev1alpha1.Aggregatable{
+							MinCount: flavour.Spec.Policy.Aggregatable.MinCount,
+							MaxCount: flavour.Spec.Policy.Aggregatable.MaxCount,
+						}
+					}
+					return nil
+				}(),
+			},
+			Owner: nodecorev1alpha1.NodeIdentity{
+				Domain: flavour.Spec.Owner.Domain,
+				IP:     flavour.Spec.Owner.IP,
+				NodeID: flavour.Spec.Owner.NodeID,
+			},
+			Price: nodecorev1alpha1.Price{
+				Amount:   flavour.Spec.Price.Amount,
+				Currency: flavour.Spec.Price.Currency,
+				Period:   flavour.Spec.Price.Period,
+			},
+			OptionalFields: nodecorev1alpha1.OptionalFields{
+				Availability: flavour.Spec.OptionalFields.Availability || false,
+			},
+		},
+		Status: nodecorev1alpha1.FlavourStatus{
+			LastUpdateTime: time.Now().String(),
+		},
+	})
+
+	return flavourCR
+}
+
+// createPeeringCandidateCustomResources creates a PeeringCandidate CR from a Flavour and a Discovery
+func createPeeringCandidateCustomResources(flavourPeeringCandidate *nodecorev1alpha1.Flavour, discovery *advertisementv1alpha1.Discovery, reserved bool, seq int) *advertisementv1alpha1.PeeringCandidate {
+	peeringCandidate := (&advertisementv1alpha1.PeeringCandidate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "peeringcandidate" + fmt.Sprintf("%d", seq),
```

Comment:
the name strategy is a little bit "crazy" 😂

- for each discovery you will start from "0"
- no unique ID
- NOT REQUIRED but, what do you think about referring to the flavour name?
- why not create a func to forge the name so that you can avoid issues?

Reply:
Name strategy less crazy now (I hope ihihih)